### ISSUE TITLE
inconsistent normalization

### ISSUE THREAD

**Issue by krukgit** [joost/phony_rails#93]
Sample uses of normalization functions:
```
> '81234567'.phony_normalized country_code: 'SG'
 => "+6581234567"
> PhonyRails.normalize_number '81234567', country_code: 'SG'
 => "+6581234567"
> Phony.normalize '81234567', cc: '65'
 => "6581234567" 
> '81234567'.phony_normalized country_code: 'SG', add_plus: false
 => "+6581234567"
> PhonyRails.normalize_number '81234567', country_code: 'SG', add_plus: false
 => "6581234567"
> PhonyRails.normalize_number '81234567', add_plus: true
 => "+81234567"
```

There are a few problems:
1. `String#phony_normalized`'s behaviour is not consistent with `PhonyRails.normalize_number`
2. `add_plus` doesn't verify whether the number includes a country code
3. `add_plus` defaults to true while `Phony#split` and `Phony#format` don't work with pluses

ad 1. 
Should be easily fixed by passing all options to `PhonyRails.normalize_number` instead of just `country_code`

ad 2.
`PhonyRails` should check whether phone number is `Phony.plausible?` before appending `+`

ad 3. 
What's the rationale behind `add_plus` defaulting to `true`? Is it to make clear the numbers are international? While it makes perfect sense, I believe it should default to `false` for the sake of compatibility with Phony. It should at least be well documented on the front page.

Let me know whether you agree with my comments and I can submit pull requests for all three points.

**Comment by krukgit** [comment]
ad 2. 
`PhonyRails.normalize_number` with `country_code` option may generate an incorrect phone number, so `Phony.plausible?` will fail even if the country code is appended. 
It seems `Phony.plausible?` is not correct for all countries, so relying on it before adding `+` would make a mess of the database.

**Comment by joost** [comment]
Regarding nr 3: it indeed is to 'know' we have a country code. There are some edge cases for which it isn't clear if the first digits is a country-code or just the plain number.
This way we differentiate between those.

**Comment by joost** [comment]
Improvement regarding 2 welcome :)

**Comment by krukgit** [comment]
I'll have to think about it. I wrote the code but then I realized it might be problematic if phone number does have a country code but is not being recognized as correct. It's ok if we assume Phony is infallible and undefined behaviour for incorrect phone numbers is fine but it might be better to just leave the responsibility in the hands of users (in that situation we need to add `add_plus` to the docs).

The code below. It breaks two tests.
https://github.com/krukgit/phony_rails/commit/0eed62fd8836cbd5daabeabcec318efb9ca39ad7
